### DISCUSSION TITLE
SU2 NEMO turbulence issues

### DISCUSSION THREAD

**Discussion by mburabay** [su2code/SU2#2379] [Q&A]
Hi,

Are any turbulence models implemented for NEMO?

I am trying to run a viscous wedge simulation with NEMO (SOLVER= NEMO_NAVIER_STOKES). Whenever I turn turbulence model on, either SA (KIND_TURB_MODEL= SA) or SST (KIND_TURB_MODEL= SST), I get a segmentation fault:

<img width="1001" alt="Screenshot 2024-11-15 at 15 22 28" src="https://github.com/user-attachments/assets/68a0080e-d44d-4165-b13c-302be0ea9cd8">

Using a bunch of cout statements, I found that error occurs on the following line, when the code tries to reach SingleGrid_Iteration:

<img width="979" alt="Screenshot 2024-11-15 at 15 26 58" src="https://github.com/user-attachments/assets/13147128-5c6a-46f9-83a4-1dc2b6a3f36f">

Then I opened a CSingleGridIntegration.cpp file, which is where the code goes from the line above when NEMO is turned off and turbulence model is turned on. It seems that the code does not enter the file (or can't find it) when NEMO and turbulence model are on. How is it possible?

If NEMO doesn't have SA and SST implemented, but the models and codes exist for other solvers, how can I implement them for NEMO?

Currently I'm using SU2 8.0.1. I know 8.1.0 was released late September but i didn't see NEMO turbulence in description of the changes.

Thanks

**Comment by bigfooted** [comment]
nemo + turbulence is not in the develop branch. There is a branch called nemo_rans that you could try. I am not sure what the status is, it looks like it stalled a bit in the last 6 months or so. Would be nice to bring it into develop. I guess the first step is to just give that version a try and see if it works for you.

**Reply by mburabay** [reply]
Thanks, I tried NEMO_RANS but it gave me an error: "SOLVER: invalid option value NEMO_RANS." I could not find the option in CConfig.cpp either. I also searched SU2 home directory and there are no matches.
